### PR TITLE
Log answers seen when submitting final answer

### LINKED ISSUE #60: Event logs should show the rationales that student see after their first submission
The course team's ability to see what the students see: what specific rationales did they see after they submit their initial response for the problem?

### PR DESCRIPTION
**Pull request by jleong-openedx**
Add answers seen to data returned on final submit

Fix test to incorporate other answers seen

Fix bug with anchoring after Step 2

Closes #60.

### CODE REVIEW

File: ubcpi/static/js/src/ubcpi.js [hunk 1]
```diff
@@ -154,6 +154,10 @@ angular.module('UBCPI', ['ngSanitize', 'ngCookies', 'gettext'])
                 self.submitting = true;
                 return backendService.submit(self.answer, self.rationale, self.status()).then(function(data) {
                     assignData(self, data);
+                    $timeout(function() {
```
Comment (jleong-openedx) [inline ubcpi/static/js/src/ubcpi.js]: Yes; this is necessary to ensure the scrolling.

File: ubcpi/ubcpi.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 MAX_RATIONALE_SIZE = 32000
 MAX_RATIONALE_SIZE_IN_EVENT = settings.TRACK_MAX_EVENT / 4
 
+otr_answers = {}
```
Comment (jleong-openedx) [inline ubcpi/ubcpi.py]: Thanks; I have changed the implementation and addressed this.

File: ubcpi/ubcpi.py [hunk 3]
```diff
@@ -548,6 +552,9 @@ def record_response(self, answer, rationale, status):
             # Send the grade
             self.runtime.publish(self, 'grade', {'value': grade, 'max_value': 1})
 
+            event_dict['other_student_responses'] = otr_answers.get("other_answers")
+
+            print(event_dict['other_student_responses'])
```
Comment (jleong-openedx) [inline ubcpi/ubcpi.py]: I've removed this now; thanks!

**Comment by jleong-openedx** [comment]
@xcompass I have updated the pull request.

File: ubcpi/ubcpi.py [hunk 4]
```diff
@@ -627,7 +627,8 @@ def get_data(self,data,suffix=''):
         """
         Retrieve persisted date from backend for current user
         """
-        return self.get_persisted_data()
+        return self.get_persisted_data(get_other_answers(
```
Comment (xcompass) [inline ubcpi/ubcpi.py]: where do you ensure the selected answers from others are the same ones that you persisted above (line 535)?

Comment (jleong-openedx) [reply ubcpi/ubcpi.py]: This is the existing way of retrieving other answers to display, but looks to be a bug, as a refresh can show a different set of answers (also tested on a version of PI prior to the commit for this pull request)--will have to implement a fix.

Comment (jleong-openedx) [reply ubcpi/ubcpi.py]: @xcompass I have implemented a fix for this, and updated the pull request.

File: ubcpi/ubcpi.py [hunk 5]
```diff
@@ -547,7 +553,7 @@ def record_response(self, answer, rationale, status):
 
             # Send the grade
             self.runtime.publish(self, 'grade', {'value': grade, 'max_value': 1})
-
+
```
[the remark this answers is not in the document]
Comment (jleong-openedx) [reply ubcpi/ubcpi.py]: Done; thanks!

**Pull request merged**